addons/account/models/ir_actions_report.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models, api, _
+from odoo.exceptions import UserError
+
+class IrActionsReport(models.Model):
+    _inherit = 'ir.actions.report'
+
+    @api.multi
+    def retrieve_attachment(self, record):
+        # get the original invoice through the message_main_attachment_id field of the record
+        if self.report_name == 'account.report_original_vendor_bill':
+            if record.message_main_attachment_id.mimetype == 'application/pdf':
+                return record.message_main_attachment_id
+            else:
+                return False
+        else:
+            return super(IrActionsReport, self).retrieve_attachment(record)
+
```

Comment:
you should also overwrite `postprocess_pdf_report` and return None in the case the report name is account.report_original_vendor_bill. Otherwise the new attachment would be saved